src/main/scala/com/microsoft/hyperspace/index/sources/default/DefaultFileBasedSource.scala

```diff
@@ -72,8 +72,20 @@ class DefaultFileBasedSource(private val spark: SparkSession) extends FileBasedS
           Hdfs.Properties(Content.fromLeafFiles(files, fileIdTracker).get)
         val fileFormatName = fileFormat.asInstanceOf[DataSourceRegister].shortName
 
+        // Store basePath of hive-partitioned data sources, if applicable.
         // "path" key in options can incur multiple data read unexpectedly.
-        val opts = options - "path"
+        // Since "options" is case-insensitive map, it will change any previous entries of
+        // "basePath" to lowercase "basepath", making it unusable.
+        // Remove lowercase "basepath" and add proper cased "basePath".
+        val opts = options.get("basePath").orElse {
+          partitionBasePath(location) match {
+            case Some(Some(path)) => Some(path)
+            case _ => None
+          }
+        } match {
+          case Some(path) => Map("basePath" -> path) ++ options - "path" - "basepath"
+          case _ => options - "path" - "basepath"
+        }
```

Comment:
For readability, 

```suggestion
        // Store basePath of hive-partitioned data sources, if applicable.
        val basePathOpt = options
          .get("basePath")
          .orElse(partitionBasePath(location).map(Some(_)))
          .map(path => "basePath" -> path)

        // "path" key in options can incur multiple data read unexpectedly.
        // Since "options" is case-insensitive map, it will change any previous entries of
        // "basePath" to lowercase "basepath", making it unusable.
        // Remove lowercase "basepath" and add proper cased "basePath".
        val opts = options - "path" - "basepath" ++ basePathOpt
```

Reply:
And I think we also need to add "basePath" option for DeltaLakeFileBasedSource.
Could you handle this for delta lake and add a test in DeltaLakeIntegrationTest likewise? Thanks!

Reply:
could you please try it once and see if the options is correctly populated with case? As I mentioned previously, this will not work because options is case insensitive.

For deltalake, I cannot take this request at this point because it's been too long for this PR and it's out of scope of this PR. I will update its description and create another issue/PR to tackle this

Reply:
https://github.com/microsoft/hyperspace/issues/295 for testing delta lake for incremental refresh

Reply:
updated the description of this PR that delta is not handled in this case.

Reply:
```suggestion
       val originalOptions =
          options.asInstanceOf[CaseInsensitiveMap[String]].originalMap

        // Extract "basePath" for partitioned source data even if it's already defined in options,
        // to get the full path of basePath.
        val basePathOpt = partitionBasePath(location).map("basePath" -> _.get)

        // "path" key in options can incur multiple data read unexpectedly.
        val opts = originalOptions - "path" ++ basePathOpt
```

Ok, I tested some and tested this code. 
1) existing basePath can be a relative path; so we need to get the full path anyway
2)  If case insensitive map is the problem, any other options including upper case should have the same problem when refreshing the index.  mergeSchema or ... etc I confirmed that `spark.read.option("basepath", "asdf").parquet("asdf/id=3")` doesn't work.

Could you apply this and revise the comment accordingly?

Reply:
thanks. I made some changes to it because of some unhandled cases.